### PR TITLE
Removed timecourse from types and added to biomaterial core. Fixes #1511

### LINKED ISSUE #1511: Add timecourse module to specimen from organism, imaged specimen, organoid biomaterials
**For which schema is a change/update being suggested?**

I would like to request updates to:

specimen_from_organism.json
imaged_specimen.json
organoid.json

*e.g.*: I would like to request an update to the `cell_line.json` schema.

**What should the change/update be?**
Requesting that "module/biomaterial/timecourse.json" be added to each of the above biomaterials

*e.g.*: I would like to add a new field - `random_integer` - to this schema to allow data contributors to assign a random integer to the cell line.

This update constitutes a minor change to the schema(s) it affects.

**Why is the change requested?**

Timecourse metadata may be relevant to certain specimens/organoids.

### PR DESCRIPTION
**Pull request by willrockout**
### Release notes

This is a remake of PR #1513 to clean up the commit tree.

For core/biomaterial/biomaterial_core.json schema:
-Added timecourse module

For type/biomaterial/donor_organism.json schema:
-Removed timecourse module

For type/biomaterial/cell_suspension.json schema:
-Removed timecourse module

For type/biomaterial/cell_line.json schema:
-Removed timecourse module

Reviews requested
-Major change, removing timecourse module from donor_organism, cell_line, and cell_suspension.
-Minor change, adding timecourse module to biomaterial core.

### CODE REVIEW

File: json_schema/core/biomaterial/biomaterial_core.json [hunk 1]
```diff
@@ -83,6 +83,12 @@
             "user_friendly": "HDBR accession",
             "example": "34526; 14758, 2, liver",
             "guidelines": "Enter accession if sample has been obtained from the Human Developmental Biology Resource (HDBR)."
+        },
+        "timecourse": {
+            "description": "Information relating to a timecourse associated with this cell line.",
```
Comment (hannes-ucsc) [inline json_schema/core/biomaterial/biomaterial_core.json:88]:
```suggestion
            "description": "Information relating to a timecourse associated with this biomaterial.",
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by amnonkhen — APPROVED** [review]

File: docs/jsonBrowser/core.md [hunk 2]
```diff
@@ -61,6 +61,7 @@ supplementary_files | A list of filenames of biomaterial-level supplementary fil
 biosamples_accession | A BioSamples accession. | string | no |  | BioSamples accession |  | SAMN00000000
 insdc_sample_accession | An International Nucleotide Sequence Database Collaboration (INSDC) sample accession. | string | no |  | INSDC sample accession |  | SRS0000000
 HDBR_accession | A Human Developmental Biology Resource (HDBR) sample accession. | string | no |  | HDBR accession |  | 34526; 14758, 2, liver
+timecourse | Information relating to a timecourse associated with this cell line. | object | no | [See module  timecourse](module.md#timecourse) | Timecourse |  |
```
Comment (hannes-ucsc) [inline docs/jsonBrowser/core.md:64]:
```suggestion
timecourse | Information relating to a timecourse associated with this biomaterial. | object | no | [See module  timecourse](module.md#timecourse) | Timecourse |  | 
```

I thought the documentation was automatically generated from the schema, so making the schema change I proposed in my previous review should have automatically caused this line to be updated, too.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
~Shouldn't this have been~

File: docs/jsonBrowser/core.md (hunk 2 above)
Comment (willrockout) [reply docs/jsonBrowser/core.md:64]: I believe it is automatically created with Git hooks upon a push but I'm not 100%. It seems to not work from a GitHub commit.

Comment (willrockout) [reply docs/jsonBrowser/core.md]: Fixed

Comment (hannes-ucsc) [reply docs/jsonBrowser/core.md]: > It seems to not work from a GitHub commit.

Well, that's a problem.

Comment (hannes-ucsc) [reply docs/jsonBrowser/core.md]: Procedural comment: The author shouldn't resolve reviewer comments. Reviewers resolve their own comments when they feel they have been addressed.

From the [SOP](https://github.com/HumanCellAtlas/dcp2/blob/main/docs/dcp2_operating_procedures.rst#id8):

> If you accept the PR author's response to a comment made by the reviewer, mark the comment thread as resolved on Github.

**Review by hannes-ucsc — APPROVED** [review]

**Review by ncalvanese1 — APPROVED** [review]

**Review by ESapenaVentura — APPROVED** [review]
LGTM!! Just a question - Based on the comments above, do I take that you manually corrected the `docs/jsonBrowser/core.md` doc?

**Comment by hannes-ucsc** [comment]
@ESapenaVentura, in my experience, commit hooks aren't a good enforcement mechanism, simply because they can easily be bypassed. What we do in my projects is make the code/doc generation part of our build process and then have a validation step in our GitHub Actions workflow that tests for a dirty working copy. If the author of a change forgets to run the code/doc generation and commit the result, the code generation will dirty the working copy on CI/CD and fail the GitHub Actions workflow. This has worked very well for us in a number of different scenarios. For example,  we also detect stale transitive Python dependencies that way.

**Comment by willrockout** [comment]
@ESapenaVentura Yes I changed it from the command line on my local copy then committed and pushed the change but for the change to the biomaterial_core.json I made the change on Git Hub.

**Pull request merged**